example/src/main/java/com/ably/chat/example/MainActivity.kt

```diff
@@ -69,29 +73,42 @@
 }
 
 @Composable
-fun Chat(chatApi: ChatApi, modifier: Modifier = Modifier) {
+fun Chat(chatClient: ChatClient, modifier: Modifier = Modifier) {
     var messageText by remember { mutableStateOf(TextFieldValue("")) }
     var sending by remember { mutableStateOf(false) }
     var messages by remember { mutableStateOf(listOf<Message>()) }
+    val listState = rememberLazyListState()
     val coroutineScope = rememberCoroutineScope()
 
     val roomId = "my-room"
+    val room = chatClient.rooms.get(roomId)
 
-    Column(
-        modifier = Modifier.fillMaxSize(),
-        verticalArrangement = Arrangement.SpaceBetween,
-    ) {
-        Button(modifier = modifier.align(Alignment.CenterHorizontally), onClick = {
+    DisposableEffect(Unit) {
+        val subscription = room.messages.subscribe {
+            messages += it.message
             coroutineScope.launch {
-                messages = chatApi.getMessages(roomId, QueryOptions(orderBy = OldestFirst)).items
+                listState.animateScrollToItem(messages.size - 1)
             }
-        }) {
-            Text("Load")
         }
 
+        coroutineScope.launch {
+            messages = subscription.getPreviousMessages().items.reversed()
+            if (messages.isNotEmpty()) listState.animateScrollToItem(messages.size - 1)
+        }
+
+        onDispose {
+            subscription.unsubscribe()
+        }
+    }
+
```

Comment:
There is no method `awaitDispose` in the `CoroutineScope`. `coroutineScope` in the code is produced by `rememberCoroutineScope` function, so it can't outlive the composable's lifecycle

Reply:
Can you make the face you made again 🙉

Reply:
Nah it was for you @coderabbitai

Reply:
Can you describe what it exactly represents @coderabbitai

Reply:
Can you add to the learnings to make happy face when you feel empathetic next time